src/backend/benchmark/tpcc/tpcc_new_order.cpp

```diff
@@ -0,0 +1,596 @@
+
+  warehouse_expr_types.push_back(
+      ExpressionType::EXPRESSION_TYPE_COMPARE_EQUAL);
+  warehouse_key_values.push_back(ValueFactory::GetSmallIntValue(warehouse_id));
+
+  auto warehouse_pkey_index = warehouse_table->GetIndexWithOid(
+      warehouse_table_pkey_index_oid);
+  
+  planner::IndexScanPlan::IndexScanDesc warehouse_index_scan_desc(
+      warehouse_pkey_index, warehouse_key_column_ids, warehouse_expr_types,
+      warehouse_key_values, runtime_keys);
+
+  // Create plan node.
+  auto predicate = nullptr;
+
+  planner::IndexScanPlan warehouse_index_scan_node(warehouse_table, predicate,
+                                                   warehouse_column_ids,
+                                                   warehouse_index_scan_desc);
+
+  executor::IndexScanExecutor warehouse_index_scan_executor(&warehouse_index_scan_node,
+                                                            context.get());
+
+  auto gwtr_lists_values = ExecuteReadTest(&warehouse_index_scan_executor);
+
+  if (txn->GetResult() != Result::RESULT_SUCCESS) {
+    txn_manager.AbortTransaction();
+    return false;
+  }
+
+  if (gwtr_lists_values.size() != 1) {
+    assert(false);
+  }
+
+  auto w_tax = gwtr_lists_values[0][0];
+
+  LOG_INFO("w_tax: %lf", w_tax.GetDouble());
+
+
+  // getDistrict
+  LOG_INFO("getDistrict: SELECT D_TAX, D_NEXT_O_ID FROM DISTRICT WHERE D_ID = ? AND D_W_ID = ?");
+
+  std::vector<oid_t> district_column_ids = {8, 10}; // D_TAX, D_NEXT_O_ID
+
+  // Create and set up index scan executor
+  std::vector<oid_t> district_key_column_ids = {0, 1}; // D_ID, D_W_ID
+  std::vector<ExpressionType> district_expr_types;
+  std::vector<Value> district_key_values;
+
+  district_expr_types.push_back(
+      ExpressionType::EXPRESSION_TYPE_COMPARE_EQUAL);
+  district_expr_types.push_back(
+      ExpressionType::EXPRESSION_TYPE_COMPARE_EQUAL);
+  district_key_values.push_back(ValueFactory::GetTinyIntValue(district_id));
+  district_key_values.push_back(ValueFactory::GetSmallIntValue(warehouse_id));
+
+  auto district_pkey_index = district_table->GetIndexWithOid(
+      district_table_pkey_index_oid);
+
+  planner::IndexScanPlan::IndexScanDesc district_index_scan_desc(
+      district_pkey_index, district_key_column_ids, district_expr_types,
+      district_key_values, runtime_keys);
+
+  // Create plan node.
+  planner::IndexScanPlan district_index_scan_node(district_table, predicate,
+                                                  district_column_ids,
+                                                  district_index_scan_desc);
+  executor::IndexScanExecutor district_index_scan_executor(&district_index_scan_node,
+                                                           context.get());
+
+  auto gd_lists_values = ExecuteReadTest(&district_index_scan_executor);
+
+  if (txn->GetResult() != Result::RESULT_SUCCESS) {
+    txn_manager.AbortTransaction();
+    return false;
+  }
+
+  if (gd_lists_values.size() != 1) {
+    assert(false);
```

Comment:
`assert` might be muted when build, and will cause `gd_lists_values` as unused variable